### PR TITLE
Remove dependency on JsonDataContractSerializer

### LINKED ISSUE #9338: Break dependency on System.Runtime.Serialization.Json
I'm investigating dependencies for .NET Core redist and noticed that the Roslyn libraries depend on System.Runtime.Serialization.Json (https://github.com/dotnet/roslyn/blob/0cee35cc375cf6a9c105b116eed776700b9f138e/src/Interactive/CsiCore/csi.rsp#L16) and the only place I see it used in the Roslyn libraries is at https://github.com/dotnet/roslyn/blob/fbae6b4d879149b42880fa2e7793ac85349f8ba8/src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs#L22. 

Based on the usage comment at https://github.com/dotnet/roslyn/blob/fbae6b4d879149b42880fa2e7793ac85349f8ba8/src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs#L97 we should consider removing this and implementing escaping locally. 

We should remove this as it does bring in a large dependency closure including System.Runtime.Serialization.Primitives, System.Runtime.Serialization.Json, System.Private.DataContractSerialization, System.Xml.XmlDocument, and System.Xml.XmlSerializer. About 1.2MB of dead weight for the usage scenario you are doing here.

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #9338 

cc @weshaggard @jaredpar @mavasani

### CODE REVIEW

File: src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs [hunk 1]
```diff
@@ -18,7 +20,7 @@ namespace Roslyn.Utilities
     /// </summary>
     internal sealed class JsonWriter : IDisposable
     {
-        private readonly StreamWriter _outptut;
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs]: I'm amazed I'm not the one to name that variable.

File: src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs [hunk 2]
```diff
@@ -146,13 +143,116 @@ private void WriteEnd(char c)
...
+            int count = 0;
+            for (int i = 0; i < value.Length; i++)
+            {
+                char c = value[i];
+
+                if (c == '\"' || c == '\'' || c == '/' || c == '\\' || ShouldAppendAsUnicode(c))
+                {
+                    if (b == null)
+                    {
+                        b = new StringBuilder(value.Length + 5);
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs]: Use `StringBuilderPool.Allocate`

File: src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs (hunk 1 above)
Comment (nguerrera) [inline src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs]: ;)

File: src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs (hunk 2 above)
Comment (nguerrera) [inline src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs]: Looks to be in Workspaces (therefore not accessible from here). Should I use PooledStringBuilder instead?

Comment (jaredpar) [inline src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs]: Yep.  Didn't realize we had two different ones.

**Comment by nguerrera** [comment]
Per offline discussion with @jaredpar and @Petermarcu, this is now blocking/urgent and will merge once CI passes. #9769 tracks follow-up work, which will be tackled later this week.

cc @mellinoe

**Comment by weshaggard** [comment]
@nguerrera can you also remove it from the project.json dependency list just to be sure there aren't any other dependencies and we don't accidentally introduce more :)

**Comment by nguerrera** [comment]
@weshaggard Will do. Tracked by #9769. Seeing some lock file weirdness (other unrelated changes on every restore) and I hear the lock files are going away and didn't want to cause merge issues. @mellimoe was convinced that not having it in assembly refs (and this removes it, I checked) is enough to unblock most urgent issue.

**Comment by weshaggard** [comment]
@nguerrera sounds good yes the main issue we are trying to unblock is the crossgening which only cares about the assembly references. We just need to make sure it doesn't show up in the list of package dependencies when you guys add netstandard support to your package as well.

**Pull request merged**